gcs/src/test/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemXAttrsIntegrationTest.java

```diff
@@ -33,38 +35,71 @@
 import java.util.EnumSet;
 import java.util.HashMap;
 import java.util.Map;
+import java.util.UUID;
 import org.apache.hadoop.fs.FileSystem;
 import org.apache.hadoop.fs.Path;
 import org.apache.hadoop.fs.XAttrSetFlag;
-import org.junit.AfterClass;
-import org.junit.BeforeClass;
+import org.junit.After;
+import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.junit.runners.JUnit4;
 
 @RunWith(JUnit4.class)
 public final class GoogleHadoopFileSystemXAttrsIntegrationTest {
 
-  private static HadoopFileSystemIntegrationHelper ghfsHelper;
-  private static FileSystem ghfs;
+  private FileSystem ghfs;
 
-  @BeforeClass
-  public static void setup() throws Throwable {
-    GoogleHadoopFileSystemIntegrationTest.storageResource.before();
-    ghfsHelper = GoogleHadoopFileSystemIntegrationTest.ghfsHelper;
-    ghfs = GoogleHadoopFileSystemIntegrationTest.ghfs;
+  private HadoopFileSystemIntegrationHelper ghfsHelper;
+
+  private GoogleCloudStorageFileSystemIntegrationHelper gcsiHelper;
+
+  private final TestBucketHelper bucketHelper =
+      new TestBucketHelper(GoogleCloudStorageIntegrationHelper.TEST_BUCKET_NAME_PREFIX);
+
+  private String bucketName;
+
+  @Before
+  public void before() throws Exception {
+
+    ghfs = new GoogleHadoopFileSystem();
+
+    URI initUri = new URI("gs://" + bucketHelper.getUniqueBucketName("init"));
+    ghfs.initialize(initUri, GoogleHadoopFileSystemTestBase.loadConfig());
+
+    gcsiHelper =
+        new GoogleCloudStorageFileSystemIntegrationHelper(
+            ((GoogleHadoopFileSystem) ghfs).getGcsFs());
+    ghfs.mkdirs(new Path(ghfs.getUri()));
+
+    ghfsHelper = new HadoopFileSystemIntegrationHelper(ghfs);
+    gcsiHelper.beforeAllTests();
+    bucketName = gcsiHelper.sharedBucketName1;
```

Comment:
It seems like this snippet of code is used in many places, should we extract it in the helper?

Reply:
Will take care of it during refactoring integration tests further.